### ISSUE TITLE
SpongeForge (Cannot push a new block change without having proxies)

### ISSUE THREAD

**Issue by yaroslav4167** [OpenMods/OpenBlocks#993]
If i placed blocks from openblocks block placed and remains in the hand.
SpongeForge: 1.12.2-2838-7.1.8-RC3967
OpenBlocks: 1.12.2-1.8.1
<details>
<summary>Log</summary>

```hs
/******************************************************************************************************************************************************************************************/
/* Proxy Access could not be popped General{Unwinding}                                                                                                                                    */
/* PhaseContext :                                                                                                                                                                         */
/*     - Source: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                              */
/*     - UnwindingState: Packet{PlaceBlock}                                                                                                                                               */
/*     - UnwindingContext: BasicPacketContext{isCompleted=true}                                                                                                                           */
/*     - IsPostingSpecial: false                                                                                                                                                          */
/*         - Owner: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                           */
/*         - Source: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                          */
/*         - PacketPlayer: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                    */
/*         - Packet: CPacketPlayerTryUseItemOnBlock{position=BlockPos{x=1279, y=71, z=1803}, placedBlockDirection=up, hand=MAIN_HAND, facingX=0.2646949, facingY=1.0, facingZ=0.12615488} */
/*         - IgnoreCreative: false                                                                                                                                                        */
/*         - ItemStackUsed: 14xtile.openblocks.vacuum_hopper@0                                                                                                                            */
/*         - OpenContainer: null                                                                                                                                                          */
/* Phases remaining :                                                                                                                                                                     */
/*   - Phase: General{Unwinding}                                                                                                                                                          */
/*     Context:                                                                                                                                                                           */
/*     - Source: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                              */
/*     - UnwindingState: Packet{PlaceBlock}                                                                                                                                               */
/*     - UnwindingContext: BasicPacketContext{isCompleted=true}                                                                                                                           */
/*     - IsPostingSpecial: false                                                                                                                                                          */
/*         - Owner: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                           */
/*         - Source: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                          */
/*         - PacketPlayer: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                    */
/*         - Packet: CPacketPlayerTryUseItemOnBlock{position=BlockPos{x=1279, y=71, z=1803}, placedBlockDirection=up, hand=MAIN_HAND, facingX=0.2646949, facingY=1.0, facingZ=0.12615488} */
/*         - IgnoreCreative: false                                                                                                                                                        */
/*         - ItemStackUsed: 14xtile.openblocks.vacuum_hopper@0                                                                                                                            */
/*         - OpenContainer: null                                                                                                                                                          */
/*   - Phase: Packet{PlaceBlock}                                                                                                                                                          */
/*     Context:                                                                                                                                                                           */
/*     - Owner: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                               */
/*     - Source: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                              */
/*     - PacketPlayer: EntityPlayerMP['Serejke_qq'/3890, l='world', x=1279.06, y=72.00, z=1805.60]                                                                                        */
/*     - Packet: CPacketPlayerTryUseItemOnBlock{position=BlockPos{x=1279, y=71, z=1803}, placedBlockDirection=up, hand=MAIN_HAND, facingX=0.2646949, facingY=1.0, facingZ=0.12615488}     */
/*     - IgnoreCreative: false                                                                                                                                                            */
/*     - ItemStackUsed: 14xtile.openblocks.vacuum_hopper@0                                                                                                                                */
/*     - OpenContainer: null                                                                                                                                                              */
/*   - Phase: Plugin{ScheduledTask}                                                                                                                                                       */
/*     Context:                                                                                                                                                                           */
/*     - Source: com.google.common.util.concurrent.ListenableFutureTask@1f4553c3                                                                                                          */
/* Stacktrace:                                                                                                                                                                            */
/* java.lang.IllegalStateException: Block Change Proxy corruption! Attempted to pop a proxy that was not on the stack.                                                                    */
/*     org.spongepowered.common.event.tracking.context.SpongeProxyBlockAccess.popProxy(SpongeProxyBlockAccess.java:128)                                                                   */
/*     org.spongepowered.common.event.tracking.context.SpongeProxyBlockAccess.access$300(SpongeProxyBlockAccess.java:60)                                                                  */
/*     org.spongepowered.common.event.tracking.context.SpongeProxyBlockAccess$Proxy.close(SpongeProxyBlockAccess.java:590)                                                                */
/*     org.spongepowered.common.event.tracking.context.MultiBlockCaptureSupplier.processTransactions(MultiBlockCaptureSupplier.java:758)                                                  */
/*     org.spongepowered.common.event.tracking.TrackingUtil.performBlockAdditions(TrackingUtil.java:591)                                                                                  */
/*     org.spongepowered.common.event.tracking.TrackingUtil.processBlockCaptures(TrackingUtil.java:460)                                                                                   */
/*     org.spongepowered.common.event.tracking.TrackingUtil.processBlockCaptures(TrackingUtil.java:389)                                                                                   */
/*     org.spongepowered.common.event.tracking.phase.packet.player.PlaceBlockPacketState.unwind(PlaceBlockPacketState.java:134)                                                           */
/*     org.spongepowered.common.event.tracking.phase.packet.player.PlaceBlockPacketState.unwind(PlaceBlockPacketState.java:67)                                                            */
/*     org.spongepowered.common.event.tracking.PhaseTracker.completePhase(PhaseTracker.java:316)                                                                                          */
/*     org.spongepowered.common.event.tracking.PhaseContext.close(PhaseContext.java:618)                                                                                                  */
/*     org.spongepowered.common.event.tracking.phase.packet.PacketPhaseUtil.onProcessPacket(PacketPhaseUtil.java:211)                                                                     */
/*     net.minecraft.network.PacketThreadUtil$1.redirect$impl$redirectToPhaseTracker$zla000(SourceFile:540)                                                                               */
/*     net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13)                                                                                                                        */
/*     java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)                                                                                                            */
/*     java.util.concurrent.FutureTask.run(FutureTask.java:266)                                                                                                                           */
/*     net.minecraft.util.Util.func_181617_a(SourceFile:46)                                                                                                                               */
/*     org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.java:308)                                                                                                   */
/*     net.minecraft.server.MinecraftServer.redirect$onRun$zje000(MinecraftServer.java:4716)                                                                                              */
/*     net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723)                                                                                                        */
/*     net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397)                                                                                              */
/*     net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)                                                                                                        */
/*     net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)                                                                                                                 */
/*     java.lang.Thread.run(Thread.java:748)                                                                                                                                              */
/*                                                                                                                                                                                        */
/* Minecraft : 1.12.2                                                                                                                                                                     */
/* SpongeAPI : 7.1.0-ea50f0c2                                                                                                                                                             */
/* Sponge : 1.12.2-7.1.8-SNAPSHOT                                                                                                                                                         */
/* SpongeForge : 1.12.2-2838-7.1.8-RC3967                                                                                                                                                 */
/* Minecraft Forge : 14.23.5.2847                                                                                                                                                         */
/******************************************************************************************************************************************************************************************/
[15:22:17] [Server thread/FATAL] [minecraft/MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Cannot push a new block change without having proxies!
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_222]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_222]
        at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
        at org.spongepowered.common.SpongeImplHooks.onUtilRunTask(SpongeImplHooks.java:308) [SpongeImplHooks.class:1.12.2-2838-7.1.8-RC3967]
        at net.minecraft.server.MinecraftServer.redirect$onRun$zje000(MinecraftServer.java:4716) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:723) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
Caused by: java.lang.IllegalStateException: Cannot push a new block change without having proxies!
        at org.spongepowered.common.event.tracking.context.SpongeProxyBlockAccess.proceed(SpongeProxyBlockAccess.java:89) ~[SpongeProxyBlockAccess.class:1.12.2-2838-7.1.8-RC3967]
        at org.spongepowered.common.event.tracking.context.BlockTransaction$ChangeBlock.enqueueChanges(BlockTransaction.java:403) ~[BlockTransaction$ChangeBlock.class:1.12.2-2838-7.1.8-RC3967]
        at net.minecraft.world.chunk.Chunk.bridge$setBlockState(Chunk.java:3234) ~[axw.class:?]
        at org.spongepowered.common.event.tracking.PhaseTracker.setBlockState(PhaseTracker.java:892) ~[(CORE)spongeforge-1.12.2-2838-7.1.8-RC3967.jar:1.12.2-2838-7.1.8-RC3967]
        at net.minecraft.world.WorldServer.func_180501_a(WorldServer.java:4519) ~[oo.class:?]
        at net.minecraft.item.ItemBlock.placeBlockAt(ItemBlock.java:180) ~[ahb.class:?]
        at openmods.item.ItemOpenBlock.func_180614_a(ItemOpenBlock.java:58) ~[ItemOpenBlock.class:?]
        at net.minecraft.item.ItemStack.func_179546_a(ItemStack.java:187) ~[aip.class:?]
        at net.minecraft.server.management.PlayerInteractionManager.func_187251_a(PlayerInteractionManager.java:957) ~[or.class:?]
        at net.minecraft.network.NetHandlerPlayServer.redirect$impl$checkState$zii000(NetHandlerPlayServer.java:2951) ~[pa.class:?]
        at net.minecraft.network.NetHandlerPlayServer.func_184337_a(NetHandlerPlayServer.java:741) ~[pa.class:?]
        at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:55) ~[ma.class:?]
        at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:11) ~[ma.class:?]
        at org.spongepowered.common.event.tracking.phase.packet.PacketPhaseUtil.onProcessPacket(PacketPhaseUtil.java:209) ~[PacketPhaseUtil.class:1.12.2-2838-7.1.8-RC3967]
        at net.minecraft.network.PacketThreadUtil$1.redirect$impl$redirectToPhaseTracker$zla000(SourceFile:540) ~[hv$1.class:?]
        at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_222]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_222]
        at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
        ... 7 more
```
</details>

**Comment by Synkstar** [comment]
Having the same issue. Its essentially a dupe

**Comment by yaroslav4167** [comment]
spongeforge-1.12.2-2838-7.1.8-RC3964 - does not have such problems.
Probably some of the versions above introduces this problem.

**Comment by boq** [comment]
Seems be related to other recently reported Sponge issue: https://github.com/OpenMods/OpenModsLib/issues/84, though it references https://github.com/SpongePowered/SpongeForge/issues/3040, which has different report.

Anyway, fix for mentioned https://github.com/SpongePowered/SpongeForge/issues/3044 seems to revert change causing those issues (https://github.com/SpongePowered/SpongeCommon/commit/9fd6f90687fa71e3839a0bb921c80994a20e6a1d#diff-0804405fdfe6e6be34e265f730b6fac4R169 vs https://github.com/SpongePowered/SpongeCommon/commit/cae8cbc6e77b5641cc26e601736bfe27743b4c21#diff-0804405fdfe6e6be34e265f730b6fac4R169), so I guess that's fixes on Sponge side (though we've also provided fix for OpenMods in https://builds.openmods.info/job/OpenModsLib/525/).

**Comment by yaroslav4167** [comment]
This bug has been fixed in new SpongeForge release 1.12.2-2838-7.1.8-RC3970.
Thank you for your attention.
